distributed/tests/test_worker.py

```diff
@@ -1807,11 +1807,23 @@ async def test_story_with_deps(c, s, a, b):
     expected_story = [
         (key, "new"),
         (key, "new", "waiting"),
+        # First log is what needs to be fetched in total as determined in
+        # ensure_communicating
         (
             "gather-dependencies",
             key,
             {fut.key for fut in futures},
         ),
+        # Second log may just be a subset of the above, see also
+        # Worker.select_keys_for_gather
+        # This case, it's all because Worker.target_message_size is sufficiently
+        # large
+        (
+            "request-dep",
+            key,
+            a.address,
+            {fut.key for fut in futures},
+        ),
```

Comment:
Since we now pass the cause properly through the methods, we also log this message properly